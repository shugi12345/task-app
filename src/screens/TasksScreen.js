import React, { useState } from 'react';
import { View, Text, TextInput, FlatList, StyleSheet, Modal, TouchableOpacity, LayoutAnimation } from 'react-native';
import DateTimePicker from '@react-native-community/datetimepicker';
import TaskItem from '../components/TaskItem';
import AsyncStorage from '@react-native-async-storage/async-storage';
import FloatingActionButton from '../components/FloatingActionButton';
import UrgencyPicker from '../components/UrgencyPicker';
import AppButton from '../components/AppButton';
import Slider from '@react-native-community/slider';

function formatDuration(mins) {
  const h = Math.floor(mins / 60);
  const m = mins % 60;
  const parts = [];
  if (h) parts.push(`${h}h`);
  if (m) parts.push(`${m}m`);
  return parts.join(' ') || '0m';
}

function parseDuration(str) {
  const match = str.match(/(?:(\d+)h)?\s*(\d+)m/);
  if (!match) return 0;
  return Number(match[1] || 0) * 60 + Number(match[2]);
}

function formatDurationWithZeros(mins) {
  const h = Math.floor(mins / 60);
  const m = mins % 60;
  const parts = [];
  if (h) parts.push(`${h}h`);
  parts.push(`${m.toString().padStart(2, '0')}m`);
  return parts.join(' ');
}

function formatDate(d) {
  return d.toISOString().slice(0, 10);
}

export default function TasksScreen() {
  const [tasks, setTasks] = useState([]);
  const [title, setTitle] = useState('');
  const [durationMinutes, setDurationMinutes] = useState(0);
  const [urgency, setUrgency] = useState(1);
  const [dueDate, setDueDate] = useState(null);
  const [showForm, setShowForm] = useState(false);
  const [editingTask, setEditingTask] = useState(null);
  const [showDatePicker, setShowDatePicker] = useState(false);
  const [showDurationPicker, setShowDurationPicker] = useState(false);

  const saveTask = () => {
    if (!title) return;

    if (editingTask) {
      const updated = tasks.map((t) =>
        t.id === editingTask.id
          ? {
              ...editingTask,
              title,
              duration: formatDuration(durationMinutes),
              urgency: Number(urgency),
              dueDate,
            }
          : t
      );
      setTasks(updated);
      AsyncStorage.setItem('tasks', JSON.stringify(updated));
    } else {
      const task = {
        id: Date.now().toString(),
        title,
        duration: formatDuration(durationMinutes),
        urgency: Number(urgency),
        dueDate,
        created: new Date(),
      };
      const newTasks = [...tasks, task];
      setTasks(newTasks);
      AsyncStorage.setItem('tasks', JSON.stringify(newTasks));
    }

    setTitle('');
    setDurationMinutes(0);
    setUrgency(1);
    setDueDate(null);
    setEditingTask(null);
    setShowForm(false);
  };

  const completeTask = (id) => {
    LayoutAnimation.configureNext(LayoutAnimation.Presets.easeInEaseOut);
    const newTasks = tasks.filter(t => t.id !== id);
    setTasks(newTasks);
    AsyncStorage.setItem('tasks', JSON.stringify(newTasks));
  };

  const startEdit = (task) => {
    setEditingTask(task);
    setTitle(task.title);
    setDurationMinutes(parseDuration(task.duration));
    setUrgency(task.urgency);
    setDueDate(task.dueDate ? new Date(task.dueDate) : null);
    setShowForm(true);
  };

  const renderItem = ({ item }) => (
    <TaskItem task={item} onPress={() => startEdit(item)} onComplete={() => completeTask(item.id)} />
  );

  return (
    <View style={styles.container}>
      <Text style={styles.appName}>Task Rabbit</Text>
      <Text style={styles.screenTitle}>Tasks</Text>
      <FlatList
        data={tasks}
        keyExtractor={item => item.id}
        renderItem={renderItem}
      />
      <FloatingActionButton onPress={() => setShowForm(true)} />

      <Modal visible={showForm} animationType="slide" transparent onRequestClose={() => setShowForm(false)}>
        <View style={styles.modalBackdrop}>
          <View style={styles.modalContent}>
            <TextInput
              placeholder="Task title"
              placeholderTextColor="#aaa"
              value={title}
              onChangeText={setTitle}
              style={styles.input}
            />
            <TouchableOpacity onPress={() => setShowDurationPicker(true)}>
              <Text style={styles.input}>{formatDuration(durationMinutes) || 'Pick duration'}</Text>
            </TouchableOpacity>
            {showDurationPicker && (
              <Modal transparent animationType="fade" onRequestClose={() => setShowDurationPicker(false)}>
                <View style={styles.modalBackdrop}>
                  <View style={styles.pickerModal}>
                    <Text style={styles.modalLabel}>{formatDurationWithZeros(durationMinutes)}</Text>
                    <Slider
                      minimumValue={0}
                      maximumValue={240}
                      step={5}
                      value={durationMinutes}
                      onValueChange={setDurationMinutes}
                      minimumTrackTintColor="#bb86fc"
                      style={{ width: '100%', height: 40, marginBottom: 16 }}
                    />
                    <AppButton title="Done" onPress={() => setShowDurationPicker(false)} />
                  </View>
                </View>
              </Modal>
            )}

            <UrgencyPicker value={urgency} onChange={setUrgency} />

            <TouchableOpacity onPress={() => setShowDatePicker(true)}>
              <Text style={[styles.input, styles.dateInput]}>
                {dueDate ? formatDate(dueDate) : 'Pick due date'}
              </Text>
            </TouchableOpacity>
            {showDatePicker && (
              <DateTimePicker
                value={dueDate || new Date()}
                mode="date"
                display="calendar"
                onChange={(e, selected) => {
                  if (selected) setDueDate(selected);
                  setShowDatePicker(false);
                }}
              />
            )}
            <View style={styles.buttonRow}>
              <AppButton title="Cancel" onPress={() => { setShowForm(false); setEditingTask(null); }} />
              <AppButton title={editingTask ? 'Save' : 'Add'} onPress={saveTask} />
            </View>
          </View>
        </View>
      </Modal>
    </View>
  );
}

const styles = StyleSheet.create({
  container: { flex: 1, padding: 20, backgroundColor: '#121212' },
  input: {
    borderColor: '#444',
    borderWidth: 1,
    marginBottom: 20,
    padding: 10,
    borderRadius: 8,
    color: '#fff',
  },
  modalBackdrop: {
    flex: 1,
    backgroundColor: 'rgba(0,0,0,0.5)',
    justifyContent: 'center',
    alignItems: 'center',
  },
  modalContent: {
    backgroundColor: '#1c1c1c',
    padding: 16,
    borderRadius: 8,
    width: '90%',
  },
  pickerModal: {
    backgroundColor: '#1c1c1c',
    padding: 16,
    borderRadius: 8,
    width: '80%',
  },
  dateInput: {
    marginTop: 20,
  },
  appName: {
    color: '#bb86fc',
    fontSize: 20,
    fontWeight: 'bold',
    marginBottom: 4,
  },
  screenTitle: {
    color: '#fff',
    fontSize: 24,
    fontWeight: 'bold',
    marginBottom: 12,
  },
  modalLabel: {
    color: '#fff',
    fontSize: 16,
    textAlign: 'center',
    marginBottom: 16,
  },
  buttonRow: {
    flexDirection: 'row',
    justifyContent: 'space-between',
    marginTop: 8,
  },
});
