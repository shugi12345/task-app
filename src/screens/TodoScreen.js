import React, { useState, forwardRef, useImperativeHandle, useEffect } from 'react';
import {
  View,
  Text,
  TextInput,
  FlatList,
  StyleSheet,
  Modal,
  TouchableWithoutFeedback,
} from 'react-native';
import TodoItem from '../components/TodoItem';
import AppButton from '../components/AppButton';
import AsyncStorage from '@react-native-async-storage/async-storage';

const TodoScreen = forwardRef((props, ref) => {
  const [items, setItems] = useState([]);
  const [text, setText] = useState('');
  const [showForm, setShowForm] = useState(false);
  const [showSortModal, setShowSortModal] = useState(false);
  const [sortMode, setSortMode] = useState('added');
  const [deletedItems, setDeletedItems] = useState([]);
  const [showHistory, setShowHistory] = useState(false);
  const textInputRef = React.useRef();

  const persistItems = (list) => {
    AsyncStorage.setItem(
      'todoItems',
      JSON.stringify(list.map(({ animateIn, ...rest }) => rest))
    );
  };

  const sortItems = (list, mode = sortMode) => {
    const sorted = [...list];
    if (mode === 'alpha') {
      sorted.sort((a, b) => a.text.localeCompare(b.text));
    } else {
      sorted.sort((a, b) => new Date(b.created) - new Date(a.created));
    }
    return sorted;
  };

  useEffect(() => {
    (async () => {
      const saved = await AsyncStorage.getItem('todoItems');
      if (saved) setItems(sortItems(JSON.parse(saved)));
      const removed = await AsyncStorage.getItem('deletedTodoItems');
      if (removed) setDeletedItems(JSON.parse(removed));
    })();
  }, []);

  useEffect(() => {
    setItems(prev => sortItems(prev, sortMode));
  }, [sortMode]);

  useImperativeHandle(ref, () => ({
    openAdd: () => setShowForm(true),
  }));

  const addItem = () => {
    if (!text) return;
    const item = { id: Date.now().toString(), text, created: new Date(), animateIn: true };
    const newItems = sortItems([...items, item]);
    setItems(newItems);
    persistItems(newItems);
    setText('');
  };

  const toggleItem = (id) => {
    const removed = items.find(i => i.id === id);
    const newItems = items.filter(i => i.id !== id);
    setItems(newItems);
    persistItems(newItems);
    if (removed) {
      const newDeleted = [removed, ...deletedItems];
      setDeletedItems(newDeleted);
      AsyncStorage.setItem('deletedTodoItems', JSON.stringify(newDeleted));
    }
  };

  const restoreItem = (id) => {
    const item = deletedItems.find(i => i.id === id);
    if (!item) return;
    const newDeleted = deletedItems.filter(i => i.id !== id);
    setDeletedItems(newDeleted);
    AsyncStorage.setItem('deletedTodoItems', JSON.stringify(newDeleted));
    const newItems = sortItems([...items, { ...item, animateIn: true }]);
    setItems(newItems);
    persistItems(newItems);
  };

  const deleteForever = (id) => {
    const newDeleted = deletedItems.filter(i => i.id !== id);
    setDeletedItems(newDeleted);
    AsyncStorage.setItem('deletedTodoItems', JSON.stringify(newDeleted));
  };

  const renderItem = ({ item }) => (
    <TodoItem item={{ ...item }} onToggle={() => toggleItem(item.id)} />
  );

  return (
    <View style={styles.container}>
      <View style={styles.headerRow}>
        <Text style={styles.screenTitle}>Todo</Text>
        <View style={styles.headerButtons}>
          <AppButton
            style={styles.sortHeaderButton}
            textStyle={styles.sortHeaderButtonText}
            title="History"
            onPress={() => setShowHistory(true)}
          />
          <AppButton
            style={styles.sortHeaderButton}
            textStyle={styles.sortHeaderButtonText}
            title={`Sort: ${sortMode === 'alpha' ? 'A-Z' : 'Latest'}`}
            onPress={() => setShowSortModal(true)}
          />
        </View>
      </View>
      <FlatList
        data={items}
        keyExtractor={item => item.id}
        renderItem={renderItem}
        contentContainerStyle={{ paddingBottom: 120 }}
      />

      <Modal
        visible={showForm}
        animationType="slide"
        transparent
        presentationStyle="overFullScreen"
        onRequestClose={() => setShowForm(false)}
      >
        <TouchableWithoutFeedback onPress={() => setShowForm(false)}>
          <View style={styles.modalBackdrop}>
            <TouchableWithoutFeedback>
              <View style={styles.modalContent}>
            <Text style={styles.modalLabel}>New Item</Text>
            <TextInput
              ref={textInputRef}
              style={styles.input}
              placeholder="Add todo"
              placeholderTextColor="#aaa"
              value={text}
              onChangeText={setText}
            />
            <View style={styles.buttonRow}>
              <AppButton
                title="Cancel"
                onPress={() => {
                  textInputRef.current?.blur();
                  setShowForm(false);
                }}
              />
              <AppButton
                title="Add"
                onPress={() => {
                  addItem();
                  textInputRef.current?.blur();
                  setShowForm(false);
                }}
              />
            </View>
              </View>
            </TouchableWithoutFeedback>
          </View>
        </TouchableWithoutFeedback>
      </Modal>

      <Modal
        visible={showHistory}
        animationType="slide"
        transparent
        presentationStyle="overFullScreen"
        onRequestClose={() => setShowHistory(false)}
      >
        <TouchableWithoutFeedback onPress={() => setShowHistory(false)}>
          <View style={styles.modalBackdrop}>
            <TouchableWithoutFeedback>
              <View style={styles.historyModal}>
            <Text style={styles.modalLabel}>Completed items</Text>
            <FlatList
              data={deletedItems}
              keyExtractor={(item) => item.id}
              renderItem={({ item }) => (
                <View style={styles.historyItem}>
                  <View style={{ flex: 1 }}>
                    <Text style={styles.historyTitle}>{item.text}</Text>
                  </View>
                  <View style={styles.buttonRow}>
                    <AppButton title="Restore" onPress={() => restoreItem(item.id)} />
                    <AppButton title="Delete" onPress={() => deleteForever(item.id)} />
                  </View>
                </View>
              )}
              ListEmptyComponent={<Text style={styles.emptyText}>No completed items</Text>}
            />
            <AppButton title="Close" onPress={() => setShowHistory(false)} />
              </View>
            </TouchableWithoutFeedback>
          </View>
        </TouchableWithoutFeedback>
      </Modal>

      <Modal
        visible={showSortModal}
        animationType="fade"
        transparent
        presentationStyle="overFullScreen"
        onRequestClose={() => setShowSortModal(false)}
      >
        <TouchableWithoutFeedback onPress={() => setShowSortModal(false)}>
          <View style={styles.modalBackdrop}>
            <TouchableWithoutFeedback>
              <View style={styles.sortModal}>
            <Text style={styles.modalLabel}>Sort items by:</Text>
            <AppButton
              style={styles.sortOption}
              title="A-Z"
              onPress={() => {
                setSortMode('alpha');
                setShowSortModal(false);
              }}
            />
            <AppButton
              style={styles.sortOption}
              title="Latest"
              onPress={() => {
                setSortMode('added');
                setShowSortModal(false);
              }}
            />
              </View>
            </TouchableWithoutFeedback>
          </View>
        </TouchableWithoutFeedback>
      </Modal>
    </View>
  );
});

export default TodoScreen;

const styles = StyleSheet.create({
  container: { flex: 1, padding: 20, backgroundColor: '#121212' },
  headerRow: {
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'space-between',
    marginBottom: 12,
  },
  headerButtons: {
    flexDirection: 'row',
    alignItems: 'center',
  },
  screenTitle: {
    color: '#fff',
    fontSize: 24,
    fontWeight: 'bold',
    marginBottom: 12,
  },
  sortHeaderButton: {
    backgroundColor: 'transparent',
    borderColor: '#bb86fc',
    borderWidth: 1,
    paddingVertical: 6,
    paddingHorizontal: 8,
    borderRadius: 8,
  },
  sortHeaderButtonText: {
    color: '#bb86fc',
    fontSize: 14,
  },
  input: {
    borderColor: '#444',
    borderWidth: 1,
    marginBottom: 20,
    padding: 10,
    borderRadius: 8,
    color: '#fff',
  },
  modalBackdrop: {
    flex: 1,
    backgroundColor: 'rgba(0,0,0,0.5)',
    justifyContent: 'center',
    alignItems: 'center',
  },
  modalContent: {
    backgroundColor: '#1c1c1c',
    padding: 16,
    borderRadius: 8,
    width: '90%',
  },
  buttonRow: {
    flexDirection: 'row',
    justifyContent: 'space-between',
    marginTop: 8,
  },
  sortModal: {
    backgroundColor: '#1c1c1c',
    padding: 16,
    borderRadius: 8,
    width: '80%',
  },
  sortOption: {
    marginVertical: 4,
  },
  historyModal: {
    backgroundColor: '#1c1c1c',
    padding: 16,
    borderRadius: 8,
    width: '90%',
    maxHeight: '80%',
  },
  historyItem: {
    flexDirection: 'row',
    alignItems: 'center',
    backgroundColor: '#1e1e1e',
    padding: 8,
    marginVertical: 4,
    borderRadius: 8,
  },
  historyTitle: {
    color: '#fff',
    fontSize: 16,
    marginRight: 8,
  },
  emptyText: {
    color: '#fff',
    textAlign: 'center',
    marginVertical: 8,
  },
  modalLabel: {
    color: '#fff',
    fontSize: 16,
    textAlign: 'center',
    marginBottom: 16,
  },
});
