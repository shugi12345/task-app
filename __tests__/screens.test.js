import React from 'react';
import renderer, { act } from 'react-test-renderer';
import TasksScreen from '../src/screens/TasksScreen';
import TodoScreen from '../src/screens/TodoScreen';

jest.mock('@react-native-async-storage/async-storage', () => require('@react-native-async-storage/async-storage/jest/async-storage-mock'));
jest.mock('@react-native-community/datetimepicker', () => 'DateTimePicker');
jest.mock('@react-native-community/slider', () => 'Slider');
jest.mock('expo-checkbox', () => 'Checkbox');

describe('Screen snapshots', () => {
  test('TasksScreen', async () => {
    let tree;
    await act(async () => {
      tree = renderer.create(<TasksScreen />);
    });
    expect(tree.toJSON()).toMatchSnapshot();
  });

  test('TodoScreen', async () => {
    let tree;
    await act(async () => {
      tree = renderer.create(<TodoScreen />);
    });
    expect(tree.toJSON()).toMatchSnapshot();
  });
});
