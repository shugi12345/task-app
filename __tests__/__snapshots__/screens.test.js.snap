// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Screen snapshots TasksScreen 1`] = `
<View
  style={
    {
      "backgroundColor": "#121212",
      "flex": 1,
      "padding": 20,
    }
  }
>
  <View
    style={
      {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "space-between",
        "marginBottom": 12,
      }
    }
  >
    <Text
      style={
        {
          "color": "#fff",
          "fontSize": 24,
          "fontWeight": "bold",
        }
      }
    >
      Tasks
    </Text>
    <View
      style={
        {
          "alignItems": "center",
          "flexDirection": "row",
        }
      }
    >
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "alignItems": "center",
            "backgroundColor": "transparent",
            "borderColor": "#bb86fc",
            "borderRadius": 8,
            "borderWidth": 1,
            "marginHorizontal": 4,
            "opacity": 1,
            "paddingHorizontal": 8,
            "paddingVertical": 6,
          }
        }
      >
        <Text
          style={
            [
              {
                "color": "#fff",
                "fontSize": 16,
                "fontWeight": "bold",
              },
              {
                "color": "#bb86fc",
                "fontSize": 14,
              },
            ]
          }
        >
          History
        </Text>
      </View>
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "alignItems": "center",
            "backgroundColor": "transparent",
            "borderColor": "#bb86fc",
            "borderRadius": 8,
            "borderWidth": 1,
            "marginHorizontal": 4,
            "opacity": 1,
            "paddingHorizontal": 8,
            "paddingVertical": 6,
          }
        }
      >
        <Text
          style={
            [
              {
                "color": "#fff",
                "fontSize": 16,
                "fontWeight": "bold",
              },
              {
                "color": "#bb86fc",
                "fontSize": 14,
              },
            ]
          }
        >
          Sort: Priority
        </Text>
      </View>
    </View>
  </View>
  <RCTScrollView
    contentContainerStyle={
      {
        "paddingBottom": 120,
      }
    }
    data={[]}
    getItem={[Function]}
    getItemCount={[Function]}
    keyExtractor={[Function]}
    onContentSizeChange={[Function]}
    onLayout={[Function]}
    onMomentumScrollBegin={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={0.0001}
    stickyHeaderIndices={[]}
    viewabilityConfigCallbackPairs={[]}
  >
    <View />
  </RCTScrollView>
</View>
`;

exports[`Screen snapshots TodoScreen 1`] = `
<View
  style={
    {
      "backgroundColor": "#121212",
      "flex": 1,
      "padding": 20,
    }
  }
>
  <View
    style={
      {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "space-between",
        "marginBottom": 12,
      }
    }
  >
    <Text
      style={
        {
          "color": "#fff",
          "fontSize": 24,
          "fontWeight": "bold",
          "marginBottom": 12,
        }
      }
    >
      Todo
    </Text>
    <View
      style={
        {
          "alignItems": "center",
          "flexDirection": "row",
        }
      }
    >
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "alignItems": "center",
            "backgroundColor": "transparent",
            "borderColor": "#bb86fc",
            "borderRadius": 8,
            "borderWidth": 1,
            "marginHorizontal": 4,
            "opacity": 1,
            "paddingHorizontal": 8,
            "paddingVertical": 6,
          }
        }
      >
        <Text
          style={
            [
              {
                "color": "#fff",
                "fontSize": 16,
                "fontWeight": "bold",
              },
              {
                "color": "#bb86fc",
                "fontSize": 14,
              },
            ]
          }
        >
          History
        </Text>
      </View>
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "alignItems": "center",
            "backgroundColor": "transparent",
            "borderColor": "#bb86fc",
            "borderRadius": 8,
            "borderWidth": 1,
            "marginHorizontal": 4,
            "opacity": 1,
            "paddingHorizontal": 8,
            "paddingVertical": 6,
          }
        }
      >
        <Text
          style={
            [
              {
                "color": "#fff",
                "fontSize": 16,
                "fontWeight": "bold",
              },
              {
                "color": "#bb86fc",
                "fontSize": 14,
              },
            ]
          }
        >
          Sort: Latest
        </Text>
      </View>
    </View>
  </View>
  <RCTScrollView
    contentContainerStyle={
      {
        "paddingBottom": 120,
      }
    }
    data={[]}
    getItem={[Function]}
    getItemCount={[Function]}
    keyExtractor={[Function]}
    onContentSizeChange={[Function]}
    onLayout={[Function]}
    onMomentumScrollBegin={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={0.0001}
    stickyHeaderIndices={[]}
    viewabilityConfigCallbackPairs={[]}
  >
    <View />
  </RCTScrollView>
</View>
`;
